### ISSUE TITLE
Unexpected exception importing platform custom_components.weatherbit.sensor

### ISSUE THREAD

**Issue by gieljnssns** [briis/weatherbit#93]
There is a new error with HA 2024.1b, weatherbit doesn't work anymore
```
Logger: homeassistant.loader
Source: loader.py:842
First occurred: December 29, 2023 at 17:09:46 (1 occurrences)
Last logged: December 29, 2023 at 17:09:46

Unexpected exception importing platform custom_components.weatherbit.sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 842, in get_platform
    cache[full_name] = self._import_platform(platform_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 859, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/config/custom_components/weatherbit/sensor.py", line 84, in <module>
    @dataclass
     ^^^^^^^^^
  File "/usr/local/lib/python3.11/dataclasses.py", line 1230, in dataclass
    return wrap(cls)
           ^^^^^^^^^
  File "/usr/local/lib/python3.11/dataclasses.py", line 1220, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dataclasses.py", line 1027, in _process_class
    _init_fn(all_init_fields,
  File "/usr/local/lib/python3.11/dataclasses.py", line 545, in _init_fn
    raise TypeError(f'non-default argument {f.name!r} '
TypeError: non-default argument 'key' follows default argument
```
```
Logger: homeassistant.setup
Source: setup.py:403
First occurred: December 29, 2023 at 17:09:46 (2 occurrences)
Last logged: December 29, 2023 at 17:10:08

Unable to prepare setup for platform 'weatherbit.sensor': Platform not found (Exception importing custom_components.weatherbit.sensor).
```
Maybe this helps https://developers.home-assistant.io/blog/2023/12/11/entity-description-changes/

**Comment by nepozs** [comment]
The version currently available as "master" seems to solve the problem (also with "stable" HA core 2024.1.0), but I will test it on a second installation.

edit: Update - Errors like:
```
Logger: homeassistant.loader
Source: loader.py:842
First occurred: 15:34:13 (1 occurrences)
Last logged: 15:34:13

Unexpected exception importing platform custom_components.weatherbit.sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 842, in get_platform
    cache[full_name] = self._import_platform(platform_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 859, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/config/custom_components/weatherbit/sensor.py", line 84, in <module>
    @dataclass
     ^^^^^^^^^
  File "/usr/local/lib/python3.11/dataclasses.py", line 1230, in dataclass
    return wrap(cls)
           ^^^^^^^^^
  File "/usr/local/lib/python3.11/dataclasses.py", line 1220, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dataclasses.py", line 1027, in _process_class
    _init_fn(all_init_fields,
  File "/usr/local/lib/python3.11/dataclasses.py", line 545, in _init_fn
    raise TypeError(f'non-default argument {f.name!r} '
TypeError: non-default argument 'key' follows default argument
```
and 
```
Logger: homeassistant.setup
Source: setup.py:403
First occurred: 15:34:13 (1 occurrences)
Last logged: 15:34:13

Unable to prepare setup for platform 'weatherbit.sensor': Platform not found (Exception importing custom_components.weatherbit.sensor).
```
has been fixed

also warning 
![warning-weatherbit-2024-01-05_18-13](https://github.com/briis/weatherbit/assets/56661567/69881022-e4fa-45ed-89e9-cccfda8c235d)
has disappeared

edit2:

Tested also with HA core 2024.1.1

Thank you @briis 
I think current version can be published (at least as beta, but most important issues are fixed, so maybe also as stable?)

**Comment by briis** [comment]
Closed with release 1.0.21
